### PR TITLE
Ability to configure HSTS response headers

### LINKED ISSUE #1793: HTTP Strict Transport Security support
Cruise Control should support HSTS in its responses for improved security.
https://https.cio.gov/hsts/

### PR DESCRIPTION
**Pull request by gergowilder**
This PR resolves #1793.

### CODE REVIEW

**Comment by viktorsomogyi** [comment]
@efeg would you please review this?

**Review by egyedt — APPROVED** [review]
LGTM! 
Thanks @gergowilder!

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java [hunk 1]
```diff
@@ -253,6 +253,29 @@
...
+  public static final boolean DEFAULT_WEBSERVER_SSL_STS_ENABLED = false;
+  public static final String WEBSERVER_SSL_STS_ENABLED_DOC = "Enables the Strict Transport Security header in the web servers responses.";
+
+  /**
+   * <code>webserver.ssl.sts.includeSubDomains</code>
+   */
+  public static final String WEBSERVER_SSL_STS_INCLUDESUBDOMAINS = "webserver.ssl.sts.includeSubDomains";
+  public static final boolean DEFAULT_WEBSERVER_SSL_STS_INCLUDESUBDOMAINS = true;
+  public static final String WEBSERVER_SSL_STS_INCLUDESUBDOMAINS_DOC = "Sets the includeSubDomains directive of the STS header. "
+    + "Only effective when webserver.ssl.sts.enabled is true.";
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java:269]: Nits: Can we use all lowercase in config names for consistency? -- i.e. `webserver.ssl.sts.includeSubDomains` -> `webserver.ssl.sts.includesubdomains` or `webserver.ssl.sts.include.subdomains`?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java [hunk 2]
```diff
@@ -253,6 +253,29 @@
...
+
+  /**
+   * <code>webserver.ssl.sts.maxAge</code>
+   */
+  public static final String WEBSERVER_SSL_STS_MAX_AGE = "webserver.ssl.sts.maxAge";
+  public static final long DEFAULT_WEBSERVER_SSL_STS_MAX_AGE = 31536000L;
+  public static final String WEBSERVER_SSL_STS_MAX_AGE_DOC = "Sets the value of the max-age (in seconds) directive of the STS header. "
+      + "Only effective when webserver.ssl.sts.enabled is true.";
+
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java:278]: Nit: Can we use all lowercase in config names for consistency?

**Review by efeg — COMMENTED** [review]
@gergowilder Thanks for the PR -- left some nits.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java (hunk 2 above)
Comment (gergowilder) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/WebServerConfig.java]: Done.

**Review by efeg — APPROVED** [review]
Thanks for the updates -- LGTM.

**Pull request merged**